client/js/render.js

```diff
@@ -35,8 +36,10 @@ function buildChatMessage(data) {
 		target = "#chan-" + chat.find(".active").data("id");
 	}
 
-	if (data.msg.preview) {
-		data.msg.preview.shown = options.shouldOpenMessagePreview(data.msg.preview.type);
+	if (data.msg.previews.length) {
+		data.msg.previews.forEach((preview) => {
+			preview.shown = options.shouldOpenMessagePreview(preview.type);
```

Comment:
You're changing how many previews are rendered though. So with this PR I assume sending 5 links will render all 5 previews at once? Seems like quite the overkill.